### ISSUE TITLE
Build issue on FreeBSD

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#98]
```
What steps will reproduce the problem?
1. Run gmake

What is the expected output? What do you see instead?
Expected leveldb to build, instead getting repeated output like

In file included from ./port/port.h:14,
                 from ./db/filename.h:14,
                 from db/builder.cc:7:
./port/port_posix.h:67: error: '__BYTE_ORDER' was not declared in this scope
./port/port_posix.h:67: error: '__LITTLE_ENDIAN' was not declared in this scope

Rolling back to 85584d497e7b354853b72f450683d59fcf6b9c5c fixes the problem.

What version of the product are you using? On what operating system?
This happens with dd0d562b4d4fbd07db6a44f9e221f8d368fee8e4

```

---

Original issue reported on code.google.com by `jhans...@gmail.com` on 20 Jun 2012 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
Oh yes, forgot to mention that this is on FreeBSD 9.0-RELEASE amd64
```

---

Original comment by `jhans...@gmail.com` on 20 Jun 2012 at 9:27

**Comment by GoogleCodeExporter** [comment]
```
Hi, in ./port/port_posix.h it compiled when I changed line 46 (__BYTE_ORDER == 
__LITTLE_ENDIAN) to (_BYTE_ORDER == _LITTLE_ENDIAN) and added #include 
<sys/endian.h>. This is also using FreeBSD 9.0-RELEASE amd64. I will do tests 
to see if it works fine.
```

---

Original comment by `wwe...@gmail.com` on 25 Jul 2012 at 3:52

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.6.0.  Please verify.

```

---

Original comment by `san...@google.com` on 12 Oct 2012 at 8:06
- Changed state: **Fixed**
